watertap/unit_models/crystallizer.py

```diff
@@ -558,6 +593,20 @@ def eq_crystallizer_height_constraint(b):
                 a + b + ((a - b) ** 2 + eps**2) ** 0.5
             )
 
+        # 13. Energy available from the vapor
+        @self.Constraint(doc="Thermal energy in the vapor")
+        def eq_vapor_energy_constraint(b):
+            return b.energy_from_vapor == (
```

Comment:
I'm thinking that there might be an IDAES standard name that can be used in place of `energy_from_vapor` or something closer to an IDAES standard if an exact one does not exist: https://idaes-pse.readthedocs.io/en/stable/explanations/conventions.html

Reply:
Also, depending on your response for the new stateblock, I wonder if `energy_from_vapor` should actually be on the property model (it seems like it should be).

Reply:
Thanks for the reference. I think renaming it to `energy_flow_superheated_vapor` or `energy_superheated_vapor` could be better.

The difference between the formula I implemented and `enth_mass_solvent` from the property model (I think you were referring to this one probably) is that I'm trying to get the energy retrieved from the condensation of a superheated steam:

E (kJ/kg) = $\lambda$(T<sub>sat</sub>) + H<sub>vap</sub>(T<sub>super</sub>) - H<sub>vap</sub>(T<sub>sat</sub>) , 
where $\lambda$ is the latent heat, T<sub>super</sub> and T<sub>sat</sub> are the temperature of superheated vapor and saturation point.

And in the property model, the `enth_mass_solvent` of the existing`properties_vapor` is:
H<sub>vap</sub>(T<sub>sat</sub>) = $\lambda$(T<sub>sat</sub>) + H<sub>liq</sub>(T<sub>sat</sub>)

Reply:
Ah, so based on this, you are essentially trying to get H<sub>vap</sub>+H<sub>liq</sub>, which sounds like it could be added to the property model perhaps as `enth_mass`. In any case, if nothing from the unit model directly impacts the constraint and it is entirely based on properties, we usually try to put that relationship in the property model. I think we should aim to do that here before we merge in.

Reply:
EDIT: maybe not Hvap-Hliq exactly, but my point being that this is something that should be resolved in the property package, not in the unit model. Is there a reason not to do so, other than "this approach works too"?

Reply:
Ohh, I got your point. I agree.

Although I think we should add `temperature_sat` as we have `pressure_sat`, and then calculate H<sub>vap</sub>(Tsat) as `enth_mass_vap_sat` or something.

Reply:
> Although I think we should add `temperature_sat` as we have `pressure_sat`, and then calculate Hvap(Tsat) as `enth_mass_vap_sat` or something.

Exactly what I was thinking too. That seems like the best way to incorporate with property model.